build.gradle.kts

```diff
@@ -145,6 +176,42 @@ subprojects {
 				}
 				jarTask.finalizedBy(extractJar)
 			}
+
+			if (project in listOf(project(":junit-platform-commons"), project(":junit-platform-console"))) {
+
+				val jarTask = (tasks.findByName("shadowJar") ?: tasks["jar"]) as Jar
+
+				jarTask.doLast {
+
+					val newFile = createTempFile("rewrite-timestamp")
+					val originalOutput = jarTask.archiveFile.get().getAsFile()
+
+					newFile.outputStream().use { os ->
+
+						val newJarStream = JarOutputStream(os)
+						val oldJar = JarFile(originalOutput)
+
+						oldJar.entries()
+								.toList()
+								.distinctBy { it.name }
+								.sortedBy { it.name }
+								.forEach { entry ->
+									val copy = JarEntry(entry.name)
+
+									// Use the same constant as the fixed timestamps in normal copy actions
+									copy.time = ZipCopyAction.CONSTANT_TIME_FOR_ZIP_ENTRIES
+
+									newJarStream.putNextEntry(copy)
+
+									oldJar.getInputStream(entry).copyTo(newJarStream)
+								}
+
+						newJarStream.finish()
+					}
+
+					newFile.renameTo(originalOutput)
+				}
+			}
```

Comment:
I'd like to move this logic to `buildSrc`. Are you feeling up for it or shall I polish it myself?

Reply:
Yep, I'll modify the commits appropriately. 👍